### PR TITLE
SITL: Make Yaw control scale with aircraft size

### PR DESCRIPTION
**Pull request by lthall**
I have approximated the yaw torque by a simple thrust vector. This is an improvement but should be replaced by a combination of propeller drag and momentum.

### CODE REVIEW

File: libraries/SITL/SIM_Motor.cpp [hunk 1]
```diff
@@ -57,9 +57,6 @@ void Motor::calculate_forces(const struct sitl_input &input,
     last_calc_us = now_us;
     last_command = command;
 
-    // the yaw torque of the motor
-    Vector3f rotor_torque = thrust_vector * yaw_factor * command * yaw_scale * voltage_scale * -1.0;
```
[the remark this answers is not in the document]
Comment (IamPete1) [reply libraries/SITL/SIM_Motor.cpp:61]: Its the same calculation in 3D to allow for none planar motors. Thrust vector is [0,0,-1] for a normal copter.

[the remark this answers is not in the document]
Comment (lthall) [reply libraries/SITL/SIM_Motor.cpp:61]: I must be missing something. It looked like everything on the right was floats.

[the remark this answers is not in the document]
Comment (lthall) [reply libraries/SITL/SIM_Motor.cpp:61]: Ahhh, thrust vector!

File: libraries/SITL/SIM_Motor.cpp [hunk 2]
```diff
@@ -72,6 +69,9 @@ void Motor::calculate_forces(const struct sitl_input &input,
     // get thrust for untilted motor
     float motor_thrust = calc_thrust(command, air_density, velocity_in, voltage_scale);
 
+    // the yaw torque of the motor
+    Vector3f rotor_torque(0, 0, yaw_factor * 0.5 * diagonal_size * yaw_scale * motor_thrust);
```
[the remark this answers is not in the document]
Comment (lthall) [reply libraries/SITL/SIM_Motor.cpp:73]: Yep, that would work. This formulation assumes the yaw torque is due to a slight angling of the motors to produce a yaw torque. Most larger aircraft need to do this so it is a part of the model we will need even when we put the motor torque in.

[the remark this answers is not in the document]
Comment (lthall) [reply libraries/SITL/SIM_Motor.cpp:73]: If we are defining thrust vector on a per motor basis the yaw torque should fall out with the right frame specification.

**Comment by lthall** [comment]
I don't understand why this is failing the guided pause test.

**Review by IamPete1 — APPROVED** [review]
Looks good. Maybe will will bring out the new scaling factor as a param configurable in the JSON file in the future.

**Pull request merged**